### PR TITLE
Attempt to switch off the Alertmanager integration for Prometheus

### PR DESCRIPTION
**Pull request by davisgomes**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go [hunk 1]
```diff
@@ -358,6 +359,13 @@ func AppendOverrides(ctx spi.ComponentContext, _ string, _ string, _ string, kvs
 		kvs = append(kvs, bom.KeyValue{Key: "prometheusOperator.tls.enabled", Value: "false"})
 	}
 
+	// Add override to switch on and off the Prometheus Alerting if Thanos is enabled
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go:362]: Minor nit, this reads a little confusing to me, how about "Add override to switch off Prometheus alerting if Thanos is enabled"?

**Review by fredtibbitts — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go (hunk 1 above)
Comment (davisgomes) [reply platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go]: Fixed!

**Review by fredtibbitts — APPROVED** [review]

File: platform-operator/thirdparty/charts/prometheus-community/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -3454,6 +3454,10 @@ prometheus:
     ##
     # podMetricsEndpoints: []
 
+  ## Allow the default Alertmanager endpoint for Prometheus.
```
Comment (desagar) [inline platform-operator/thirdparty/charts/prometheus-community/kube-prometheus-stack/values.yaml:3457]: minor nit - Enable or Create rather than Allow

**Review by desagar — DISMISSED** [review]

File: platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go [hunk 3]
```diff
@@ -543,6 +551,51 @@ func appendAdditionalVolumeOverrides(ctx spi.ComponentContext, volumeMountKey, v
...
 }
 
+// appendAlertmanagerIntegrationOverrides disables the Alertmanager integration with Prometheus if Thanos Ruler is enabled
+func appendAlertmanagerIntegrationOverrides(ctx spi.ComponentContext, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
+	rulerEnabled, err := isThanosRulerEnabled(ctx)
+	if err != nil {
+		return kvs, err
+	}
+	return append(kvs, []bom.KeyValue{
+		{Key: "prometheus.enableDefaultAlertingEndpoint", Value: strconv.FormatBool(!rulerEnabled)},
```
[the remark this answers is not in the document]
Comment (davisgomes) [reply platform-operator/controllers/verrazzano/component/prometheus/operator/promoperator.go:562]: Thanks. I added these to the keys.

File: platform-operator/thirdparty/charts/prometheus-community/kube-prometheus-stack/values.yaml (hunk 2 above)
Comment (davisgomes) [reply platform-operator/thirdparty/charts/prometheus-community/kube-prometheus-stack/values.yaml]: Fixed

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**